File: sion_fasion_shop/ProyectoMVC/SionFashioWebApplicationMVC/SionFashioWebApplicationMVC/Models/tienda.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SionFashioWebApplicationMVC.Models;

public partial class tienda
{
    [Key]
    public int id_tienda { get; set; }

    [Required(ErrorMessage = "El nombre de la tienda es obligatorio.")]
    [StringLength(255, ErrorMessage = "El nombre de la tienda no puede exceder los 255 caracteres.")]
    public string nombre_tienda { get; set; } = null!;

    [StringLength(255, ErrorMessage = "El teléfono de la tienda no puede exceder los 255 caracteres.")]
    [RegularExpression(@"^\+?\d+$", ErrorMessage = "El teléfono de la tienda solo puede contener números positivos y el símbolo '+'.")]
    public string? telefono_tienda { get; set; }

    [StringLength(255, ErrorMessage = "La ubicación de la tienda no puede exceder los 255 caracteres.")]
    public string? ubicacion_tienda { get; set; }

    [Required(ErrorMessage = "El ID de la ciudad es obligatorio.")]
    public int id_ciudad { get; set; }

    [Required(ErrorMessage = "El ID de la empresa es obligatorio.")]
    public int id_empresa { get; set; }


    [InverseProperty("id_tiendaNavigation")]
    public virtual ICollection<empleado> empleados { get; set; } = new List<empleado>();

    [ForeignKey("id_ciudad")]
    [InverseProperty("tienda")]
    public virtual ciudade id_ciudadNavigation { get; set; } = null!;

    [ForeignKey("id_empresa")]
    [InverseProperty("tienda")]
    public virtual empresa id_empresaNavigation { get; set; } = null!;

    [InverseProperty("id_tiendaNavigation")]
    public virtual ICollection<producto> productos { get; set; } = new List<producto>();
}

File: sion_fasion_shop/Entregables/Trimestre V/Sistema de inventario en Node.Js/Bck-End SionFashion-Shop/models/tienda.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

// Esquema del modelo Tienda
const tiendaSchema = new Schema({

    nombre_tienda: {
        type: String,
        required: [true, 'El nombre de la tienda es obligatorio.'],
        maxlength: [255, 'El nombre de la tienda no puede exceder los 255 caracteres.']
    },
    telefono_tienda: {
        type: String,
        maxlength: [255, 'El teléfono de la tienda no puede exceder los 255 caracteres.'],
        match: [/^\+?\d+$/, 'El teléfono de la tienda solo puede contener números positivos y el símbolo "+".']
    },
    ubicacion_tienda: {
        type: String,
        maxlength: [255, 'La ubicación de la tienda no puede exceder los 255 caracteres.']
    },
    id_ciudad: {
        type: Schema.Types.ObjectId,
        required: [true, 'El ID de la ciudad es obligatorio.'],
        ref: 'Ciudade' // Asegúrate de tener el modelo "Ciudad" definido
    },
    id_empresa: {
        type: Schema.Types.ObjectId,
        required: [true, 'El ID de la empresa es obligatorio.'],
        ref: 'Empresa' // Asegúrate de tener el modelo "Empresa" definido
    },
    empleados: [{
        type: Schema.Types.ObjectId,
        ref: 'Empleado' // Asegúrate de tener el modelo "Empleado" definido
    }],
    productos: [{
        type: Schema.Types.ObjectId,
        ref: 'Producto' // Asegúrate de tener el modelo "Producto" definido
    }]
});

// Crea el modelo Tienda
const Tienda = mongoose.model('Tienda', tiendaSchema);

module.exports = Tienda;
